controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java

```diff
@@ -283,19 +283,18 @@ public Void initializeStreamWriters(final String commitStreamName, final EventSt
      * @param scope               scope name.
      * @param stream              stream name.
      * @param lease               txn lease.
-     * @param maxExecutionPeriod  maximum amount of time for which txn may remain open.
      * @param scaleGracePeriod    amount of time for which txn may remain open after scale operation is initiated.
      * @param ctx                 context.
      * @return                    identifier of the created txn.
      */
     CompletableFuture<Pair<VersionedTransactionData, List<Segment>>> createTxnBody(final String scope,
                                                                                    final String stream,
                                                                                    final long lease,
-                                                                                   final long maxExecutionPeriod,
                                                                                    final long scaleGracePeriod,
                                                                                    final OperationContext ctx) {
         // Step 1. Validate parameters.
-        CompletableFuture<Void> validate = validate(lease, maxExecutionPeriod, scaleGracePeriod);
+        CompletableFuture<Void> validate = validate(lease, scaleGracePeriod);
+        long maxExecutionPeriod = MAX_EXECUTION_TIME_MULTIPLIER * lease;
```

Comment:
Let's top it at 24 hours to avoid insanely large execution times.

Reply:
done